src/cpdetect/heuristics/pe_heuristics.cpp

```diff
@@ -937,7 +937,7 @@ void PeHeuristics::getUpxHeuristics()
 	// format: x.xx'\0'UPX!
 	const std::size_t minPos = 5, verLen = 4;
 	pos = content.find("UPX!");
-	if (pos >= minPos && pos < 0x500 && pos < sections[0]->getOffset())
+	if (pos >= minPos && pos < 0x500 && sections.size() > 0 && pos < sections[0]->getOffset())
```

Comment:
Consider replacing `sections.size() > 0` with `!sections.empty()` (we want to check if a vector is empty, which can be done directly using `std::vector::empty()`).

Reply:
This is a good idea. Will fix it right away.